FontNode.js

```diff
@@ -0,0 +1,50 @@
+import path from 'path';
+import './lib/inflate.js';
+import './lib/unbrotli.js';
+import './Font.js';
+import fs from 'fs';
+import { getFontCSSFormat } from './Font.js';
+
+fs.readFileAsync = (filename) => {
+    return new Promise((resolve, reject) => {
+        fs.readFile(filename, (err, data) => {
+            if (err) reject(err);
+            else resolve(data);
+        });
+    });
+};
+
+export default class FontNode extends Font {
+    constructor(fileRaw, options = {}) {
+        if (fileRaw) {
+            const file = path.normalize(path.resolve(fileRaw)) || '';
+            const name = file.replace(/^.*\/(.*?)(?:\..*?)$/, '$1');
+            const font = super(name, options);
+            font.src = file;
+            return new Promise(function (resolve, reject) {
```

Comment:
I'm confused about the `src` setter, it calls `async loadFile` but doesn't handle the promise at all. How is it supposed to be used? Looks like it should be removed.

Reply:
We still need `src` to fall through to loadFile, as it's different from what the superclass does, but it doesn't need to handle the promise: if you want to set the `src` (i.e. you want to use the exact same code for browser and node) then that should be possible, but then the signal mechanism is going to be the `onload` and `onerror` functions.

Reply:
Indeed, this way it is similar to the regular Font class. Will take a look at it.

Reply:
@Pomax right, I didn't look into the context of the superclass, makes more sense now. However, I'm thinking about Nodejs `UnhandledPromiseRejectionWarning` which references [DEP0018: Unhandled promise rejections ](https://nodejs.org/dist/latest-v8.x/docs/api/deprecations.html#deprecations_dep0018_unhandled_promise_rejections):

> Type: Runtime
>
> Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

Here's a demo how the `src` setter will fail in Nodejs:

```js
class Font {
  set src(url){ this._raiser();}
  async _raiser(){throw new Error('Houston, we have a problem')}
}
font = new Font()
font.src = 'my source path';
> (node:1620) UnhandledPromiseRejectionWarning: Error: Houston, we have a problem
    [... stack traceback ...]
(node:1620) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 4)
```

From the context you provided I'd expect something like this, though, in this example we would get duplicated calls to `this.onerror`.

```js
set src(url) {
    this.__src = url;
    this.loadFile(url)
    .catch(error=>{
        if (!this.onerror) throw err;
        this.onerror(err);;
    })
  }
```

To not have the duplicate `this.onerror` invocations, an option would be not to `throw err;` when `this.onerror` is available, same as I did in `set src`:

```js
  async loadFile(url) {
    this.__src = url; // because this needs to work as direct call, too.

    const file = path.normalize(path.resolve(url));

    if (!file) {
      const err = new Error(...bad file path: not found...)
      if (!this.onerror) throw err;
      this.onerror(err);
      return;
    }

    const ext = path.extname(file);

    if (!ext) {
      const err = new Error(...bad file path: missing extension...)
      if (!this.onerror) throw err;
      this.onerror(err);
      return;
    }

    // Note that getFontCSSFormat relies on a prefix `.`
    // which `path.extname` conveniently leaves in.
    const type = getFontCSSFormat(ext);

    if (!type) {
      const err = new Error(...unknown file type error...)
      if (!this.onerror) throw err;
      this.onerror(err);
      return;
    } 

    return  fs.readFile(url)
      .then(buffer => new Uint8Array(buffer).buffer)
      .then(arrayBuffer => this.fromDataBuffer(arrayBuffer, type));
  }
```

Reply:
Also the promise from `.then(arrayBuffer => this.fromDataBuffer(arrayBuffer, type));` in `loadFile` should be handled the same as in `Font.loadFont(url)`: https://github.com/Pomax/Font.js/blob/44c95880b5876e1a95ce59ed1aea21ddbc12f437/Font.js#L105-L110

Reply:
it shouldn't, because the `loadFile` function can be called by anyone as per the suggested updated docs - loadFile can be used with "your own" `.then` and `.catch` handling - so if the unhandled `catch` in `src` is no longer supported, we can add it there:

```
set src(url) {
  this._src = url;
  this.loadFile(url).catch(e => this.onerror ? this.onerror(e) : ());
}
```

Reply:
I think I'd prefer an  `async setSrc` method over `set src` plus event handling to write code that works in the browser and node. But I guess I don't get the merits of having event handling attached to this. ;-)